### ISSUE TITLE
How to access services in Symfony4?

### ISSUE THREAD

**Issue by ThomasLandauer** [Codeception/Codeception#4976]
Following up on https://github.com/Codeception/Codeception/issues/4752 , how can I access services in Symfony4?

* First attempt, using `grabService()`:

`$this->tester->grabService('App\Service\CsvImporter');` gives me:
> Service \App\Service\CsvImporter is not available in container

* Second attempt, using Codeception's depency injection from https://codeception.com/docs/07-AdvancedUsage#Dependency-Injection :
```php
protected function _inject(App\Service\CsvImporter $csvImporter)
{
    $this->csvImporter = $csvImporter;
}
```
gives me:
> [InjectionException] Failed to inject dependencies in instance of 'CsvImporterTest'. Failed to create instance of 'App\Service\CsvImporter'. Failed to resolve dependency 'Doctrine\ORM\EntityManagerInterface'.

This is because the service has `EntityManagerInterface` injected.

**Comment by Besedin86** [comment]
@ThomasLandauer  I found solution for Symfony >= 4.1 
Just write helper method and use it to get Symfony service in your tests.

```php
public function getSymfonyService(string $id)
{
    return $this->getModule('Symfony')->grabService('test.service_container')->get($id);
}
```

**Comment by bogdan-dubyk** [comment]
I have a similar issue. I add `getSymfonyService ` to functional Helper as mentioned by @Besedin86. 

In my symfony service.yml  I have:


```
    App\ApiClient:
    App\HttpClientInterface: '@App\ApiClient'

```

But when I run `$this-tester->getSymfonyService(HttpClientInterface::class);` I'm getting

> Service App\HttpClientInterface is not available in container

**Comment by Besedin86** [comment]
@bogdan-dubyk which version of Symfony do you use?

**Comment by bogdan-dubyk** [comment]
@Besedin86  4.1.4

**Comment by bogdan-dubyk** [comment]
Nm, It's working now, I think it was cache

**Comment by bogdan-dubyk** [comment]
Okay, but what if I want to associate some mock with service? how can I add mocked object to service container? 

I tried:

```
    public function replaceSymfonyServiceWithMock(MockInterface $object, string $alias)
    {
        /** @var Symfony $symfony */
        $symfony = $this->getModule('Symfony');
        $testContainer = $symfony->grabService('test.service_container');

        $testContainer->set($alias, $object);
        $symfony->persistService($alias, false);
    }
```

but it's not working. Also, I tried to put it in real container:

  ```
 public function replaceSymfonyServiceWithMock(MockInterface $object, string $alias)
    {
        /** @var Symfony $symfony */
        $symfony = $this->getModule('Symfony');
        $testContainer = $symfony->grabService('kernel')->getContainer();

        $testContainer->set($alias, $object);
        $symfony->persistService($alias, false);
    }

```

but I'm getting the error 

>  service is private, you cannot replace it.

**Comment by StalkAlex** [comment]
@bogdan-dubyk have you solved somehow this problem?

**Comment by ThomasLandauer** [comment]
Fixed in Codeception 3.1.2 via https://github.com/Codeception/Codeception/pull/5706
